src/collects/seashell/backend/asan-error-parse.rkt

```diff
@@ -0,0 +1,129 @@
+#lang typed/racket
+
+;; Use the function "asan-error-parse" to reads the asan error message
+;; and procude a parsed structure
+(provide asan-rewrite)
+
+(struct StackFrame ([frame : Integer]
+                    [file : String]
+                    [ln : Integer]
+                    [func : String])
+  #:transparent)
+(define-type ErrorComponent (U String StackFrame))
+(define-type ASANError (Listof ErrorComponent))
+
+;; wrapper function
+(: asan-rewrite (Bytes -> Bytes))
+(define (asan-rewrite x)
+  (cond [(zero? (bytes-length x)) x]
+        [else (string->bytes/utf-8
+                (format "\n~a\n" 
+                        (ASANError->string 
+                          (asan-error-parse (bytes->string/utf-8 x)))))]))
+
+;; rewrite ASANError in a friendly way
+(: ASANError->string (ASANError -> String))
+(define (ASANError->string x)
+  (string-join 
+    (filter-map 
+      (lambda ([x : ErrorComponent])
+        (match x
+          [(pregexp #px"^=+\\d+=+ERROR: AddressSanitizer: SEGV") "Error: Segmentation fault"]
```

Comment:
Can you grab more detailed error information?  Ideally you could replicate the functionality from https://github.com/cs136/seashell/blob/master/src/frontend/frontend/console.js#L65.